File: code/cpp/class_struct.cpp
// BEGIN_CODE
class MyClass {
private:
	const int my_field;

public:
	// constructor
	MyClass(const int &field) : my_field(filed) { }

	// destructor
	~MyClass() { }

	// friend functions
	ostream& operator<<(ostream &out, const MyClass &my_class) {
		out << my_field;
	}
}
// END_CODE

File: code/rust/class_struct.rs
// BEGIN_CODE
struct MyClass {
	my_field: isize
}

impl MyClass {
	// constructor, by convention
	pub fn new(field: isize) -> MyClass {
		MyClass { my_field: field }
	}
}

impl Drop for MyClass {
	// destructor
	fn drop(&mut self) {
	}
}
// END_CODE
